app/src/main/java/com/yalta/repositories/UserRepo.kt

```diff
@@ -5,17 +5,15 @@ import common.User
 import okhttp3.OkHttpClient
 import okhttp3.Request
 
-sealed class UserResponse
-class GotUser(val user: User) : UserResponse()
-class NoUser : UserResponse()
+class GotUser(val user: User) : SuccessfulResponse()
 
 interface UserRepo {
-    suspend fun getUser(id: Long): UserResponse
+    suspend fun getUser(id: Long): RepoResponse
```

Comment:
Yeah, seems too generic... This should be more specific... Hmm...

Maybe 
```
sealed ResponseType()
Success : ResponseType()
Failed: ResponseType()

Response(val type:ResponseType)
UserResponse : Response(...)
...
```
Then we are still able to pattern-match but also narrow return values as much as possible?

Reply:
OR
```
sealed class Response<S>(val content:S)
class SuccessfulResponse<S>:Response<S>()
class FailedResponse<S>:Response<S>
...
class UserResponse(){}
...
getUser(): Response<UserResponse>
```
We can play with naming here so that `Response<something>` could be read like english

Reply:
OR maybe we can just self-type? (can we?)
```
sealed class Response<S>()
class SuccessfulResponse<S>:Response<S>()
class FailedResponse<S>:Response<S>
...
class UserReponse : SuccessfullResponse<UserResponse>
```
Or this can be an alias?! hmmm
then we can define:
```
suspend fun getUser(id: Long): Response<UserResponse>
```
we can still pattern-match cause `Response<UserReponse>` is both `FailedResponse<UserResponse>` and `Successful<UserResponse>` AND last one is just UserResponse!